### PR TITLE
gha: upload code coverage to codecov.io

### PR DESCRIPTION
**Pull request by dtrifiro**

### CODE REVIEW

**Comment by dtrifiro** [comment]
please note that CODECOV_TOKEN will have to be added to the repo's secrets.

See here https://app.codecov.io/gh/opendatahub-io/model-registry/new

**Comment by dtrifiro** [comment]
@isinyaaa

File: .github/workflows/python-client.yml [hunk 1]
```diff
@@ -63,14 +63,13 @@ jobs:
...
 
-      - name: Upload coverage data
-        if: always() && matrix.session == 'tests'
-        uses: "actions/upload-artifact@v3"
-        with:
-          name: coverage-data
-          path: "clients/python/.coverage.*"
+       name: Upload coverage report
+        uses: codecov/codecov-action@v3.1.4
+        if: always()
```
Comment (antisaling) [inline .github/workflows/python-client.yml]: IDK if we should enable cov generation when building docs, I think the original condition here was right.

```diff
-        if: always()
+        if: always() && matrix.session == 'tests'
```

**Review by antisaling — APPROVED** [review]
lgtm

**Comment by rareddy** [comment]
Thank you @dtrifiro 

Would love to see the Go tests reported along side this!

**Comment by lampajr** [comment]
> Thank you @dtrifiro
> 
> Would love to see the Go tests reported along side this!

@rareddy I created this issue for the Golang part https://github.com/opendatahub-io/model-registry/issues/170

**Comment by rareddy** [comment]
Thank you @lampajr

**Pull request merged**